packages/samples/test/output/rest/petstore/openapi.json

```diff
@@ -184,10 +184,10 @@
         "description": "Creates or update a instance of the extension resource.",
         "parameters": [
           {
-            "$ref": "#/components/parameters/Pet.petId"
+            "$ref": "#/components/parameters/Pet.id"
           },
           {
-            "$ref": "#/components/parameters/Checkup.checkupId"
+            "$ref": "#/components/parameters/Checkup.id"
```

Comment:
Hmm, in the diff I saw in chat, things got inlined which seemed fine, but now I'm wondering if referencing Checkup.id for parameter named checkupId is OK. Is it possible to produce a repro where Checkup.id is used as a parameter directly and as a key parameter with a different name than it had orinally and in that case will we hit the duplicate name error?

Reply:
message after, the change was to include the sourceProperty in `copyKeysProperties` which then cause the openapi emitter logic to keep tracing the property back to its source